vertical-pod-autoscaler/builder/Dockerfile

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.19.5
+FROM golang:1.20.2
```

Comment:
@voelzmo Open question: Do we want to restrict golang Docker image bumps to patch versions only?